src/main/java/be/syntra/devshop/DevshopFront/services/CartServiceImpl.java

```diff
@@ -15,18 +15,27 @@
 import javax.annotation.PostConstruct;
 import java.math.BigDecimal;
 import java.util.List;
+import java.util.stream.Collectors;
 
 @Slf4j
 @Service
 public class CartServiceImpl implements CartService {
     private CartDto currentCart;
+    private ProductService productService;
```

Comment:
these fields can all be final

Reply:
**correct me if I'm wrong. But haven't I said this once already? Were there some problems concerning the datastore? Don't know anymore.

Reply:
made them final, about your question: i'm afraid i have no idea anymore.
the dataStore is only used for remembering where to go when the login process is finished, so i guess you're talking about the @component annotation?

Reply:
could be. Now, as long as you've tested the application 'functionally', and it works, then I'm perfectly happy.
I was just thinking, when the class was loaded when it was still a bean, the final field would have prevented the second bean initialisation. Component shouldn't have that problem.